.github/workflows/build-and-test-devcontainer.yml

```diff
@@ -0,0 +1,88 @@
+name: Build and Test DevContainer
+
+on:
+  push:
+    paths:
+      - '.devcontainer/'
+  pull_request:
+    paths:
+      - '.devcontainer/'
+  workflow_dispatch:
+
+jobs:
+  shellcheck-and-test:
+
+    name: Shellcheck linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup PATH
+        run: |
+          echo ::notice title={Setup Permissions}::{Add permissions to .devcontainers/}
+          chmod -R +rwx .devcontainer
+          
+          echo ::notice title={Setup Workspace}::{Add .devcontainers/ to PATH}
+          echo "${{github.workspace}}/.devcontainer" >> "$GITHUB_PATH"
+          echo "${{github.workspace}}/.devcontainer/tests" >> "$GITHUB_PATH"
+
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
+        env:
+          SHELLCHECK_OPTS: -s bash
+        with:
+          format: tty
+          severity: error
+          additional_files: post_start_cli_autocomplete.sh
+          scandir: "${{github.workspace}}/.devcontainer"
+        continue-on-error: true
+
+      - name: Setup Bats and bats libs
+        id: setup-bats
+        uses: bats-core/bats-action@3.0.0
+
+      - name: Test Devcontainer scripts
+        shell: bash
+        env:
+          BATS_LIB_PATH: ${{ steps.setup-bats.outputs.lib-path }}
+        run: bats .devcontainer/post_start_cli_autocomplete_test.bats
+
+  build-with-devcontainer-cli:
+    name: Devcontainer CLI Build and Push Cache
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup PATH
+        run: |
+          echo ::notice title={Setup Permissions}::{Add permissions to .devcontainers/}
+          chmod -R +rwx .devcontainer
+
+          echo ::notice title={Setup Workspace}::{Add .devcontainers/ to PATH}
+          echo "${{github.workspace}}/.devcontainer" >> "$GITHUB_PATH"
+          echo "${{github.workspace}}/.devcontainer/tests" >> "$GITHUB_PATH"
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - run: npm install -g @devcontainers/cli
```

Comment:
I'm not 100% sure on this, but I think `npm ci` is safer for use in CI environments as it should result in reproducible installs based on the lock file. Where-as `npm install` is better for installing locally for the first time as it won't install reproducibly on the first install.

Reply:
Yea that's totally a good callout thanks!

Reply:
Ah so `npm ci` is dependent on a lockfile being present for a node project.  Node in this case is only used during the build process with the devcontainer CLI but isn't naturally in the `.devcontainer` folder.

I'll find a better way to manage this in my fast-follow PR.